### DISCUSSION TITLE
no G7 pairing request on Huawei P30

### DISCUSSION THREAD

**Discussion by kameamea** [NightscoutFoundation/xDrip#2942] [Q&A] [answered]
When I start a G7 sensor with the patched Dexcom app i see a Bluetooth pairing request after perhaps one minute or two, then can start the sensor and later let xdrip connect to it.

However when i try to start the sensor with xdrip directly I don't get this pairing request and thus only receive mismatch errors.

I remember having issues with this phone with G6 pairing as well until finally some xdrip Bugfix solved it. 

Device: Huawei P30
xdrip version: 2023.06.08
There were a few differences to the recommended settings, regarding bluetooth watchdog and calibration. I changed those now and will try again with the next sensor in 9 days.

**Comment by Navid200** [comment, marked as answer]
Are you using the latest Nightly?
Are you using the recommended settings?  https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html

Thanks for opening this discussion.  No, please don't open an issue yet.

**Reply by kameamea** [reply]
Today with update to version from 2023-07-16 and using recommended settings
I started a new sensor.
This time i received the pairing request. Good. But unfortunately I got another problem now while or after the warmup period:
An invalid timestamp comparison message, see screenshot:
![Screenshot_20230724_154809_com eveningoutpost dexdrip](https://github.com/NightscoutFoundation/xDrip/assets/7243765/c6638dc2-d65a-407b-af2e-d20953b81eea)

I then tried to start the same sensor with the Dexcom app, but that reported a sensor failure message, so i had to replace the new sensor.I don't know whether  xdrip caused this failure, hopefully you can make sense out of this screenshot?
In the meantime I removed the sensor and found the filament hidden in the small hole of the sensor. So it never went through my skin as it looks. 
Anyway the timestamp message from xdrip looks strange to me.

**Reply by jamorham** [reply]
If the sensor physically hasn't deployed properly then its quite likely that we might get strange data from it that could produce the error messages seen. I'm interested to know whether a good sensor you are able to get work correctly.

**Reply by kameamea** [reply]
Today i inserted a new sensor and everything worked well (I again rebooted the phone before, next time I'll try without that). Thanks for all that great work!

**Comment by Navid200** [comment]
And please always include the phone brand and model in a post like this.  So, please edit your first post and include those pieces of info.
